File: 品味Java的21種設計模式/dp_java/mediator/example2/VideoCard.java
package cn.javass.dp.mediator.example2;
/**
 * 显卡类，一个同事类
 */
public class VideoCard extends Colleague{
	public VideoCard(Mediator mediator) {
		super(mediator);
	}
	
	/**
	 * 显示视频数据
	 * @param data 被显示的数据
	 */
	public void showData(String data){
		System.out.println("您正观看的是："+data);
	}
	
}

File: 品味Java的21種設計模式/dp_cpp/mediator/example2/VideoCard.cpp
#include "VideoCard.h"

namespace cn {
namespace javass {
namespace dp {
namespace mediator {
namespace example2 {

VideoCard::VideoCard(Mediator *mediator) : Colleague(mediator) {
}

void VideoCard::showData(std::string data) {
    puts("您正观看的是：" + data);
}
}
}
}
}
}
